include/drivers/flash_partition.h

```diff
@@ -0,0 +1,231 @@
+ */
+struct flp_info_cfg {
+	const struct device *device;
+	const struct flp_info *parent;
+	const off_t off;
+	const size_t size;
+	const bool read_only;
+};
+
+/**
+ * @brief flp_info_state represent the state of a partition.
+ */
+struct flp_info_state {};
+
+/**
+ * @brief flp_info represents a partition.
+ */
+struct flp_info {
+	const struct flp_info_cfg *cfg;
+	const struct flp_info_state *state;
+};
+
+/**
+ * @brief Read data from flash partition. Read boundaries are verified before
+ * read request is executed.
+ *
+ * @param[in]  flp Flash partition
+ * @param[in]  off Read offset from beginning of flash partition
+ * @param[out] dst Buffer to store read data
+ * @param[in]  len Number of bytes to read
+ *
+ * @return  0 on success, negative errno code on fail.
+ */
+int flp_read(const struct flp_info *flp, off_t off, void *dst, size_t len);
+
+/**
+ * @brief Write data to flash partition. Write boundaries are verified before
+ * write request is executed.
+ *
+ * @param[in]  flp Flash partition
+ * @param[in]  off Write offset from beginning of flash partition
+ * @param[out] src Buffer with data to be written
+ * @param[in]  len Number of bytes to write
+ *
+ * @return  0 on success, negative errno code on fail.
+ */
+int flp_write(const struct flp_info *flp, off_t off, const void *src,
+	      size_t len);
+
+/**
+ * @brief Erase range on flash partition. Erase boundaries are verified before
+ * erase is executed.
+ *
+ * @param[in] flp Flash partition
+ * @param[in] off Erase offset from beginning of flash partition
+ * @param[in] len Number of bytes to be erase (should be a multiple of
+ *                erase-block-size)
+ *
+ * @return  0 on success, negative errno code on fail.
+ */
+int flp_erase(const struct flp_info *flp, off_t off, size_t len);
+
+/**
+ * @brief Get flash parameters (write-block-size and erase-value) of the device
+ * the partition is on.
+ *
+ * @param[in] flp Flash partition
+ *
+ * @return 0 on success, negative errno code on fail.
+ */
+const struct flash_parameters *flp_get_parameters(const struct flp_info *flp);
+
+/**
+ * @brief Retrieve a flash partition layout.
+ *
+ * A flash partition layout is a run-length encoded description of the
+ * pages on the partition.
+ *
+ * For flash memories which have uniform page sizes, this routine
+ * returns an array of length 1, which specifies the page size and
+ * number of pages in the partition.
+ *
+ * Layouts for flash partitions with nonuniform page sizes will be
+ * returned as an array with multiple elements, each of which
+ * describes a group of pages that all have the same size. In this
+ * case, the sequence of array elements specifies the order in which
+ * these groups occur on the partition.
+ *
+ * @param[in] flp Flash partition to retrieve the layout from
+ * @param[out] layout Flash layout
+ * @param[out] layout_size Number of elements in the layout
+ */
+void flp_pages_layout(const struct flp_info *flp,
```

Comment:
this method is not implemeted

Reply:
This is a left-over when I tried to wrap this into a flash_device. As it is not possible to do the above function I needed to remove it. Thanks for spotting this.

Reply:
I removed the method.